### ISSUE TITLE
Does not work with nested fragments where one cannot match

### ISSUE THREAD

**Issue by CodeWitchBella** [dotansimha/graphql-code-generator#3064]
**Describe the bug**
When I have fragment which includes nested `... on Type` in it and use that fragment on type which does not implement `Type` codegen crashes.
 

**To Reproduce**

https://codesandbox.io/s/graphql-codegen-issue-template-57mxc

1. My GraphQL schema:

```graphql
interface Venue {
    id: String!
    name: String!
}

type GPSPosition {
    lat: Float!
    lng: Float!
}

interface VenueWithPosition {
    id: String!
    gpsPosition: GPSPosition!
}

type Hotel implements VenueWithPosition & Venue {
    id: String!
    gpsPosition: GPSPosition!
    name: String!
}

type Transport implements Venue {
    id: String!
    name: String!
}

type Query {
    hotel: Hotel!
    transport: Transport!
}

```

2. My GraphQL operations:

```graphql
  fragment venue on Venue {
    id
    ... on VenueWithPosition {
      gpsPosition {
        lat
        lng
      }
    }
  }

  query q {
    hotel {
      ...venue
    }
    transport {
      ...venue
    }
  }
```

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
documents: document.tsx
generates:
  types.tsx:
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
      - fragment-matcher

```

**Expected behavior**
Does not crash and generates output

**Environment:**

- OS: Linux
- `@graphql-codegen/...`: 1.9.1
- NodeJS: 12.13.0

**Additional context**

Works in apollo codegen

**Comment by dotansimha** [comment]
@CodeWitchBella Thank you for reporting it. 
It caused by the fact that codegen expects inline fragments to match the actual types. In your case, `Transport` will never have `gpsLocation` field, so codegen fails because of that, even though it's a valid syntax. 

I managed to reproduce this issue in this PR: https://github.com/dotansimha/graphql-code-generator/pull/3204

Can you please try the alpha version? `1.9.2-alpha-f55b39c1.100`

**Comment by dotansimha** [comment]
Fixed in `v1.10.0`.
